jax2onnx/converter/ir_postprocess.py

```diff
@@ -0,0 +1,275 @@
+                continue
+            name = _value_name(out)
+            if name and name in io_names:
+                continue
+            new_shape = _make_unknown_shape_like(
+                getattr(out, "shape", None), force_rank_only=force_rank_only
+            )
+            if new_shape is not None:
+                out.shape = new_shape
+                tensor_type = getattr(out, "type", None)
+                if isinstance(tensor_type, ir.TensorType):
+                    out.type = ir.TensorType(tensor_type.elem_type)
+            if name:
+                seen.add(name)
+
+    # Clean up any existing value_info containers if present
+    for attr_name in ("value_info", "_value_info"):
+        vi_list = getattr(graph, attr_name, None)
+        if isinstance(vi_list, list):
+            filtered: list[ir.Value] = []
+            for vi in vi_list:
+                name = _value_name(vi)
+                if name and name in io_names:
+                    filtered.append(vi)
+                    continue
+                new_shape = _make_unknown_shape_like(
+                    getattr(vi, "shape", None), force_rank_only=force_rank_only
+                )
+                if new_shape is not None:
+                    vi.shape = new_shape
+                    tt = getattr(vi, "type", None)
+                    if isinstance(tt, ir.TensorType):
+                        vi.type = ir.TensorType(tt.elem_type)
+                filtered.append(vi)
+            setattr(graph, attr_name, filtered)
+
+
+def _tensor_to_numpy(tensor_obj: object) -> np.ndarray | None:
+    if tensor_obj is None:
+        return None
+    if hasattr(tensor_obj, "to_numpy"):
+        try:
+            return tensor_obj.to_numpy()
+        except Exception:
+            pass
+    try:
+        return np.asarray(tensor_obj)
+    except Exception:
+        return None
+
+
+def _maybe_promote_value_to_double(val: ir.Value) -> None:
+    const_val = getattr(val, "const_value", None)
+    arr = _tensor_to_numpy(const_val)
+    if arr is None or arr.dtype != np.float32:
+        return
+    promoted = ir.tensor(arr.astype(np.float64))
+    val.const_value = promoted
+    if isinstance(getattr(val, "type", None), ir.TensorType):
+        val.type = ir.TensorType(ir.DataType.DOUBLE)
+    else:
+        val.type = ir.TensorType(ir.DataType.DOUBLE)
+
+
+def _promote_constant_attributes(node: ir.Node) -> None:
+    attrs = _attribute_iter(node)
+    for attr in attrs:
+        attr_name = getattr(attr, "name", None)
+        attr_type = getattr(attr, "type", None)
+        if attr_name != "value" or attr_type != IRAttrType.TENSOR:
+            continue
+        tensor_obj = getattr(attr, "value", None)
+        arr = _tensor_to_numpy(tensor_obj)
+        if arr is None or arr.dtype != np.float32:
+            continue
+        promoted_tensor = ir.tensor(arr.astype(np.float64))
+        if hasattr(attr, "_value"):
+            attr._value = promoted_tensor
```

Comment:
Setting a private field is not supported. Normally we consider Attr to be immutable and should be replaced if modified